nav2-usability/comparative-analysis/README.md

```diff
@@ -0,0 +1,38 @@
+# Flutter Router Package Comparative Analysis  
+As part of the [Navigator 2.0 API Usability Research](https://github.com/flutter/uxr/wiki/Navigator-2.0-API-Usability-Research), we have conducted a [preliminary comparative analysis](https://github.com/flutter/uxr/issues/13) of existing router packages Flutter community has produced around common navigation scenarios ([PDF](https://github.com/flutter/uxr/files/5953028/PUBLIC.Flutter.Navigator.API.Scenarios.-.Storyboards.pdf)) the Flutter UX Research team has identified through previous user studies.  
+  
+## Analysis  
+Scroll right to see all >
+
+  | auto_route | beamer | flit-router | flouter | fluro | flutter_modular | page_router | qlevar_router | routemaster | vrouter
+-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
+Deep Linking - Path Parameters | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅
+Deep Linking - Query Parameters | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | 🆇 | ✅ | ✅ | ✅
+Dynamic Linking | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅
+Validation (Login/Logout/Sign-up Routing) | ✅ - route validation happens at the route pattern level, but no custom validators (e.g. check if id is valid before routing) | ✅ - BeamGuard | 🆇 - in TODO list | 🆇 | 🆇 - in TODO list | ✅ | ✅ - no way to specify "default" route when validation fails (e.g. go to login/ screen if user is logged out) | ✅ | ✅ | ✅- VNavigationGuard
+Nested Routing | ✅ | 🆇 - in TODO list. Currently supports nested routing, but it's not yet synchronized with browsers | 🆇 - in TODO list (highest priority) | 🆇 - in TODO list | 🆇 - in TODO list | ✅ | 🆇 | ✅ | ✅ | ✅
+Skipping Stacks | ✅ | ✅ | ✅ | ✅ | 🆇 | 🆇 | ✅ | ✅ | ✅ | ✅
+Custom Pages / Custom transition animations | ✅ | ✅ - slovnicki/beamer#51 (TODO: slovnicki/beamer#108) | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅- Custom transitions are supported, but custom Page objects aren't (it currently hard-codes MaterialPage)
+
+## Packages Included in the Analysis  
+Learn more about the packcages inluded in this analysis on Github:  
+  
+[auto_route](https://github.com/Milad-Akarie/auto_route_library)
+[beamer](https://github.com/slovnicki/beamer)
+[flit-router](https://github.com/polyflection/flit_router)
+[flouter](https://github.com/Kleak/flouter)
+[fluro](https://github.com/lukepighetti/fluro)
+[flutter_modular](https://github.com/Flutterando/modular)
+[page_router](https://github.com/johnpryan/page_router)
+[qlevar_router](https://github.com/SchabanBo/qlevar_router)
+[routemaster](https://github.com/tomgilder/routemaster)
+[vrouter](https://github.com/lulupointu/vrouter)
+ 
+## Acknowledgment  
+Authors of the abovementioned packages shared their feedback on the analysis via email and Github [comments](https://github.com/flutter/uxr/issues/13). We appreciate your contributions.  
+  
+## This is a living document
```

Comment:
It would be great if we can link to example code from the table for different package and scenario combinations. Maybe we can do that for a few packages we're familiar with and ask authors to update the table for the rest.